ledger/daml-on-sql/src/test/suite/scala/on/sql/CliSpec.scala

```diff
@@ -42,6 +45,26 @@ class CliSpec
       config shouldEqual None
     }
 
+    "reject when sql-backend-jdbcurl-env is not a PostgreSQL URL" in {
+      fakeEnv += "JDBC_URL" -> "jdbc:h2:mem"
+      val config =
+        cli.parse(Array("--ledgerid", "test-ledger", "--sql-backend-jdbcurl-env", "JDBC_URL"))
+      fakeEnv -= "JDBC_URL"
+      config shouldEqual None
+    }
+
+    "accept a PostgreSQL JDBC URL from sql-backend-jdbcurl-env" in {
```

Comment:
Can you add a test to ensure that it fails when the environment variable is not set? I think right now it's a no-op, which is not what I'd expect.

Reply:
Actually, I meant it like this, i.e. if it's not set it will fall back on using the one on command line (if provided, else it will fail). Do you think it's tricky?

Reply:
I think it could be confusing and possibly dangerous. If I make a typo in the environment variable name, falling back is definitely _not_ what I'd expect.

Reply:
Yes, I agree, especially considering this is a security feature. I'll add the check.

Reply:
Done.